agent/src/datafile.rs

```diff
@@ -435,6 +435,36 @@ impl DataFile {
         self.store(inner);
     }
 
+    /**
+     * Mark a particular running snapshot as failed to provision.
+     */
+    pub fn fail_rs(&self, region_id: &RegionId, snapshot_name: &str) {
+        let mut inner = self.inner.lock().unwrap();
+
+        let mut rs = inner
+            .running_snapshots
+            .get_mut(region_id)
+            .unwrap()
```

Comment:
The agent would panic, and the service would restart, scan the data file, and attempt to do the same operation. It may then succeed, or it may crash again. Too many crashes would put the service in a maintenance state and it remain disabled.

Reply:
Do we know for sure that a panic'd agent will restart and retry?  I know there are places where we don't 
correctly handle panics, but I was not sure what happens to the agent.